sysboundary/outflow.cpp

```diff
@@ -359,54 +359,40 @@ namespace SBC {
-         && !this->facesToSkipFields[2]                     // we are not supposed to skip fields on this face
-         && component == 1                                  // we do the component normal to this face
-         && !(isThisCellOnAFace[0] || isThisCellOnAFace[1] || isThisCellOnAFace[4] || isThisCellOnAFace[5]) // we are not in a corner
-      ) {
-         propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, false, true, false);
-         if(RKCase == RK_ORDER1 || RKCase == RK_ORDER2_STEP2) {
-            fieldValue = perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
-         } else {
-            fieldValue = perBDt2Grid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
-         }
-      } else if (sysBoundaryLayer == 1
-         && isThisCellOnAFace[4]                            // we are on the face
-         && this->facesToProcess[4]                         // we are supposed to do this face
-         && !this->facesToSkipFields[4]                     // we are not supposed to skip fields on this face
-         && component == 2                                  // we do the component normal to this face
-         && !(isThisCellOnAFace[0] || isThisCellOnAFace[1] || isThisCellOnAFace[2] || isThisCellOnAFace[3]) // we are not in a corner
-      ) {
-         propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, false, false, true);
-         if(RKCase == RK_ORDER1 || RKCase == RK_ORDER2_STEP2) {
-            fieldValue = perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
-         } else {
-            fieldValue = perBDt2Grid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
+         if (neighborSysBoundaryFlag == sysboundarytype::NOT_SYSBOUNDARY) {
+            switch(component) {
+               case 0:
+                  propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, true, false, false);
+                  break;
+               case 1:
+                  propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, false, true, false);
+                  break;
+               case 2:
+                  propagateMagneticField(perBGrid, perBDt2Grid, EGrid, EDt2Grid, i, j, k, dt, RKCase, false, false, true);
+                  break;
+               default:
+                  break;
+            }
+            if(RKCase == RK_ORDER1 || RKCase == RK_ORDER2_STEP2) {
+               fieldValue = perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
+            } else {
+               fieldValue = perBDt2Grid.get(i,j,k)->at(fsgrids::bfield::PERBX + component);
+            }
+            return fieldValue;
          }
+      }
+
+      if(RKCase == RK_ORDER1 || RKCase == RK_ORDER2_STEP2) {
+         fieldValue = fieldBoundaryCopyFromExistingFaceNbrMagneticField(perBGrid, technicalGrid, i, j, k, component);
       } else {
```

Comment:
Instead of searching for the nearest neighbor, should we just choose the cell directly now that we know it's a sysboundarycell and we know the i,j,k to it and the component?

Reply:
But we don't know where that nearest neighbour is (corner cases).

Reply:
Ah, right we want to copy from the nearest neighbor *non*-sysboundarycell?

Reply:
As the code and documentation clearly state. Not.
https://github.com/fmihpc/vlasiator/blob/ea5a3345f45ee542432e10d498bf8a9973fb65b6/sysboundary/sysboundarycondition.cpp#L798